### ISSUE TITLE
X.grad is none

### ISSUE THREAD

**Issue by ReemAlsharabi** [facebookresearch/CrypTen#488]
I'm trying to get the gradients of cryptensor but it's returning None:
```
images_enc = crypten.cryptensor(images, requires_grad=True)
labels_enc = crypten.cryptensor(labels, requires_grad=True)

def train_model(model, X, y, epochs=10, learning_rate=0.05):
    criterion = crypten.nn.CrossEntropyLoss()

    for epoch in range(epochs):
        model.zero_grad()
        output = model(X)
        loss = criterion(output, y)
        print(f"epoch {epoch} loss: {loss.get_plain_text()}")
        loss.backward()
        grads = X.grad
        print(grads)
        model.update_parameters(learning_rate)
    return model, grads

model, grads = train_model(model, images_enc[:10, ], labels_enc[:10,], epochs=3)
```

Output of print(grads): `None`

**Comment by ReemAlsharabi** [comment]
The problem was using [:10, ] indexing. See #210 
Sol.:
replace `model, grads = train_model(model, images_enc[:10, ], labels_enc[:10,], epochs=3)` 
with `model, grads = train_model(model, images_enc, labels_enc, epochs=3)`
